app/workers/asset_manager_update_asset_worker.rb

```diff
@@ -12,10 +12,11 @@ def perform(attachment_data, legacy_url_path, new_attributes = {})
     rescue GdsApi::HTTPNotFound
       asset_missing = true
     end
+    asset_deleted = attributes['deleted']
 
-    if asset_missing && attachment_data.deleted?
+    if (asset_missing || asset_deleted) && attachment_data.deleted?
       return
-    elsif asset_missing && !attachment_data.deleted?
+    elsif (asset_missing || asset_deleted) && !attachment_data.deleted?
```

Comment:
@chrisroos Again, I think we should raise an exception if the asset is missing, shouldn't we? i.e. maybe we no longer need to rescue `GdsApi::HTTPNotFound` and set `asset_deleted`.

Reply:
I've made some further changes to `AssetManagerUpdateAssetWorker#perform` so that it no longer rescues `GdsApi::HTTPNotFound` exceptions.